src/components/measureGroups/PopulationHelperFunctions.tsx

```diff
@@ -0,0 +1,158 @@
+import * as _ from "lodash";
+
+export enum InitialPopulationAssociationType {
+  DENOMINATOR = "Denominator",
+  NUMERATOR = "Numerator",
+}
+
+export const addPopulation = (insertCallback, scoring, population, index) => {
+  //scoring and population
+  // check scoring, population type and then invoke insert on index
+  let secondAssociation = undefined;
+  if (scoring === GroupScoring.RATIO) {
+    // const ip = populations[populationIndex];
+    const ip = population;
+    if (ip?.associationType === InitialPopulationAssociationType.DENOMINATOR) {
+      secondAssociation = InitialPopulationAssociationType.NUMERATOR;
+    } else if (
+      ip?.associationType === InitialPopulationAssociationType.NUMERATOR
+    ) {
+      secondAssociation = InitialPopulationAssociationType.DENOMINATOR;
+    }
+  }
+  insertCallback(index + 1, {
+    id: uuidv4(),
+    name: population.name,
+    definition: "",
+    associationType: secondAssociation,
+  });
+};
+
+export const showAddPopulationLink = (
+  scoring: string,
+  populations: Population[],
+  population: Population
+) => {
+  // For ratio group
+  if (scoring === GroupScoring.RATIO) {
+    const ips = findPopulations(populations, PopulationType.INITIAL_POPULATION);
+    // if IP is one, add second IP link is shown
+    return (
+      ips &&
+      ips.length === 1 &&
+      population.name === PopulationType.INITIAL_POPULATION
+      // ips.name === PopulationType.INITIAL_POPULATION
+    );
+  }
+  return false;
+};
+
+export const removePopulationCallback = (removeCallback) => {
+  removeCallback;
+};
+
+export const findIndex = (
+  population: Population,
+  populations: Population[]
+) => {
+  for (let index = 0; index < populations?.length; index++) {
+    const temp = populations[index];
+    if (population.id === temp.id) {
+      return index;
+    }
+  }
+};
+export const changeAssociation = (
+  populationIndex,
+  scoring,
+  populations,
+  population,
+  replaceCallback,
+  setAssociationChanged
+) => {
+  if (scoring === GroupScoring.RATIO) {
+    const ips = findPopulations(populations, PopulationType.INITIAL_POPULATION);
+    if (ips && ips.length === 2) {
+      ips.forEach((ip) => {
+        if (ip.id !== population.id) {
+          if (
+            population.associationType ===
+            InitialPopulationAssociationType.DENOMINATOR
+          ) {
+            ip.associationType = InitialPopulationAssociationType.NUMERATOR;
+          } else if (
+            population.associationType ===
+            InitialPopulationAssociationType.NUMERATOR
+          ) {
+            ip.associationType = InitialPopulationAssociationType.DENOMINATOR;
+          }
+          const index = findIndex(ip, populations);
+          replaceCallback(index, ip);
+          setAssociationChanged(true);
+        }
+      });
+    } else {
+      replaceCallback(populationIndex, population);
+    }
+  }
+};
+
+export const getInitialPopulationSize = (scoring, populations) => {
```

Comment:
I have no idea. 
I didn't change the logic on these. Just made the functions take the arguments they were referencing instead of depending on scope and put them in a file.